jobs/sweep_schedulers/optuna_scheduler/optuna_scheduler.py

```diff
@@ -465,6 +468,40 @@ def _get_run_history(self, run_id: str) -> List[int]:
         history = api_run.scan_history(keys=["_step", metric_name])
         metrics = [x[metric_name] for x in history]
 
+        if len(metrics) == 0 and api_run.lastHistoryStep > -1:
+            logger.debug("No metrics, but lastHistoryStep exists")
+            wandb.termwarn(
+                f"{LOG_PREFIX}Detected logged metrics, but none matching " +
+                f"provided metric name: '{metric_name}'"
+            )
+            if self._try_infer_metric:
+                try:
+                    api_run.load(force=True)
+                except Exception as e:
+                    logger.debug(f"Failed to force load run from public api: {str(e)}")
+                    self._try_infer_metric = False
+                    return []
+
+                run_keys = api_run._attrs["historyKeys"]["keys"].keys()
+                potential_keys = [
+                    k
+                    for k in run_keys
+                    for kwrd in ["loss", "acc", "metric", "score"]
+                    if kwrd in k
+                ]
+                if len(potential_keys) > 0:
+                    metric_name = potential_keys[0]
+                    wandb.termwarn(
+                        f"{LOG_PREFIX}Inferring metric name from logged metrics, now" +
+                        f" using metric: '{metric_name}'. To disable this behavior, " +
+                        "set a correct metric.name or set scheduler.settings._try_" +
+                        "infer_metric to false in the sweep config."
+                    )
+                    self._sweep_config["metric"]["name"] = metric_name
+
+                # only try inference once, even if it fails, don't try again
+                self._try_infer_metric = False
```

Comment:
this is sort of crazy, happy to delete this.

Reply:
Yeah, just throw an error IMO, because you also need to have inversion on some of these right? Like acc wants ot be maxed, loss minned